plugins/homekit/src/types/camera/camera-streaming.ts

```diff
@@ -355,14 +355,11 @@ export function createCameraStreamingDelegate(device: ScryptedDevice & VideoCame
                 let rtspServer: RtspServer;
                 let track: string;
                 let playing = false;
-                session.audioReturn.once('message', async buffer => {
-                    try {
-                        const decrypted = srtpSession.decrypt(buffer);
-                        const rtp = RtpPacket.deSerialize(decrypted);
-
-                        if (rtp.header.payloadType !== session.startRequest.audio.pt)
-                            return;
+                let initializing = false;
```

Comment:
This code seems correct, but it was a bit confusing to read due to the addition of a state. Can you make a change with something like:

```ts
let twoWayAudioState = 'stopped' | 'starting' | 'started' = 'stopped';
```

Then use this variable throughout to manage check change/check.

Reply:
Related: this would seem to allow restarting of the two way audio if it terminates for some reason. Incidentally, I may consider terminating two way audio automatically if homekit stops sending packets (which it does when the mic is disabled on the ios home app). Many cameras operate with echo cancellation during two way audio, which mutes incoming audio to some degree.

Reply:
Done. Much easier to read, I agree. And yes, it does allow restarting of the two way audio!